trade-finance-manager-api/src/graphql/reducers/mappings/facilities/mapUkefExposure.js

```diff
@@ -1,17 +1,47 @@
+const { getUnixTime } = require('date-fns');
 const { formattedNumber } = require('../../../../utils/number');
+const api = require('../../../../v1/api');
+const { calculateNewFacilityValue, calculateUkefExposure, latestAmendmentValueAccepted } = require('../../../helpers/amendment.helpers');
 
-const mapUkefExposure = (facilityTfm) => {
-  const {
-    ukefExposure,
-    ukefExposureCalculationTimestamp,
-  } = facilityTfm;
+// maps ukef exposure on original value or latest amended value
+const mapUkefExposure = async (facilityTfm, facility) => {
+  if (facilityTfm) {
+    const {
+      ukefExposure,
+      ukefExposureCalculationTimestamp,
+      exchangeRate,
+    } = facilityTfm;
 
-  const formattedUkefExposure = formattedNumber(ukefExposure);
+    // sets facility exposure values and timestamp
+    let formattedUkefExposure = formattedNumber(ukefExposure);
+    let ukefExposureCalculationTimestampValue = ukefExposureCalculationTimestamp;
 
-  return {
-    exposure: `GBP ${formattedUkefExposure}`,
-    timestamp: `${ukefExposureCalculationTimestamp}`,
-  };
+    if (facility && facility?._id) {
+      const { _id } = facility;
+
+      const latestCompletedAmendment = await api.getLatestCompletedAmendment(_id);
+
+      if (latestCompletedAmendment?.amendmentId && latestCompletedAmendment?.value && latestAmendmentValueAccepted(latestCompletedAmendment)) {
```

Comment:
can we add this condition in a constant? there are a few conditions here

Reply:
done

Reply:
and can we follow the comments that I've added in the other files?
i.e. we don't need `latestCompletedAmendment?.amendmentId`